File: Server/Game/RankManager.cs
using Server.Model;
namespace Server.Game

{

    public static class RankManager
    {
        private static readonly Dictionary<string, RankInfo> _ranks = new();

        public static RankInfo Get(string playerId)
        {
            if (!_ranks.ContainsKey(playerId))
                _ranks[playerId] = new RankInfo { PlayerId = playerId };
            return _ranks[playerId];
        }

        public static void AddWin(string playerId)
        {
            var info = Get(playerId);
            info.Wins++;
            info.Score += 15;
        }

        public static void AddLoss(string playerId)
        {
            var info = Get(playerId);
            info.Losses++;
            info.Score -= 10;
            if (info.Score < 0) info.Score = 0;
        }
    }
}

File: Server/src/RankManager.cpp
#include "Game/RankManager.h"

using namespace Server;
using namespace System::Collections::Generic;

RankManager::RankManager()
{
    ranks = gcnew Dictionary<String^, RankInfo^>();
}

void RankManager::UpdateRank(String^ playerName, bool win)
{
    if(!ranks->ContainsKey(playerName))
        ranks[playerName] = gcnew RankInfo(playerName);

    if(win) ranks[playerName]->AddWin();
    else ranks[playerName]->AddLoss();
}

RankInfo^ RankManager::GetRank(String^ playerName)
{
    if(ranks->ContainsKey(playerName))
        return ranks[playerName];
    return nullptr;
}
